### PR TITLE
Adding functions

### PR DESCRIPTION
**Pull request by danielamoreno699**
In this project I added logic to the calculator component. For that purpose, I created a new folder or directory, called logic. where I inserted the calculator.js and also the operate.js. The first file, No only contains the structure of the calculator component, but it also handles the states, which are transfered as parameters to buttom component and the screen component. 
On the other hand, the operate.js file, handles all the logic of the functions of the calculator. I created multiple functions for each button and at the same time, they update the state of the calculator when the it is clicked.  I used the following state :  
[calc, setCalc] = useState({
    operator: '',
    operand1: 0,
    operand2: 0,
    result: 0,
    screenValue: '',
  });'  , so each function updates the corresponding result.
the math logic is stored in the function operator, which taked the operands and the operator in order to execute the math operation inside the respective functions

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @danielamoreno699,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Your README is missing a lot sections listed on the Table of Contents. You have no corresponding sections for Install, Usage, Run tests, etc. When I click the links they go nowhere. Kindly ensure all the links in the Table of Contents have corresponding sections (only FAQ is optional).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/logic/calculator.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  });
+
+  return (
+    <div className="wrapper">
+      <Screen value={calc.screenValue} />
+      <ButtonBox calc={calc} setCalc={setCalc} operate={operate} />
+    </div>
+  );
+};
+export default Calculator;
```
Comment (Whoistolu) [inline src/components/logic/calculator.js]: - [ ] [REQUIRED] To make work easy and concentrate on how to work with React events, you are advised to use the already developed math logic needed to make the Calculator component functional. Kindly ensure that you use this file as recommended [calculate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js). Please copy the file into your logic directory then analyze the file and be sure you understand what they do.

- Kindly refer to the project requirement:
Screenshot 2023-04-27 at 17 22 51

File: src/components/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  const newNum = -num;
+
+  setCalc({
+    ...calc,
+    screenValue: newNum.toString(),
+    operand1: calc.operator ? parseFloat(calc.operand1) : newNum,
+    operand2: calc.operator ? newNum : parseFloat(calc.operand2),
+    result: calc.operator ? operate(calc.operator, parseFloat(calc.operand1), newNum) : calc.result,
+  });
+};
```
Comment (Whoistolu) [inline src/components/logic/operate.js]: - [ ] [REQUIRED] To make work easy and concentrate on how to work with React events, you are advised to use the already developed math logic needed to make the Calculator component functional. Kindly ensure that you use this file as recommended [operate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js). Please copy the file into your logic directory then analyze the file and be sure you understand what they do.

- Kindly refer to the project requirement:
Screenshot 2023-04-27 at 17 22 51

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @danielamoreno699 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

### N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator-components/children.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/* eslint-disable-line react/no-array-index-key */
```
Comment (Ghiftee) [inline src/components/calculator-components/children.js:1]: - [OPTIONAL] I suggest that you do not disable linters as they help us to write cleaner code

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @danielamoreno699, 

To highlight 👍🏽 
✔️ App runs as expected
✔️ Props are passed
✔️ Logic files are created

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**